### ISSUE TITLE
Issue in conversion of Map to JSONObject in buildJSON method

### ISSUE THREAD

**Issue by cmijum** [cloudant/sync-cordova-plugin#90]
Hi Team I find an issue in the code where the Map.Entry<String, Object> is converted to a JSONObject inside the buildJSON function of CloudantSyncPlugin.java

```
for (Map.Entry<String, Object> entry : body.entrySet()) {
            result.put(entry.getKey(), entry.getValue());
}
```

Lets take a multi-level json 
```
{
  "_id": "96dbfe3da7d543b106ba6ca324ca5c2d",
  "_rev": "1-44ab60d541e09bf132206e8553b31621",
  "document_type": "Vehicle",
  "vehicle_types": [
    {
      "type": "Car",
      "model": "4 Wheeler"
    },
    {
      "type": "Bike",
      "model": "2 Wheeler"
    }
  ]
}
```
Code that fetches the result : 
```
let query = {
              selector: {
                            document_type: "Vehicle",
                        }
            };        
this.datastore.find(query)
       .then(function(results){
                console.log("Results ",results);                         
       }).done();
```
If you look at the result of it
![image](https://user-images.githubusercontent.com/14971151/28953750-e34c7fa6-78f7-11e7-869b-dbdbbcb38725.png)

It has not converted the child entries (vehicle_types) to a proper Json Object.
Convering this 

> "[{type=Car, model=4 Wheeler}, {type=Bike, model=2 Wheeler}]"

 back to a json object is again a lengthy process.
Can someone look into this issue and get it fixed

Thanks & Regards,
Miju

**Comment by ricellis** [comment]
Thank you for reporting this issue, I have reproduced the problem and we will work on the fix.

**Comment by cmijum** [comment]
Hi @ricellis, Thank you for responding.
I made a temporary fix on my local by replacing

 ```
for (Map.Entry<String, Object> entry : body.entrySet()) {
           result.put(entry.getKey(), (Object)entry.getValue());
 }

```
with 
```
JSONObject objBody = new JSONObject(body);        
for(Iterator<String> keys=objBody.keys();keys.hasNext();) {
     String key = keys.next();
     result.put(key, objBody.get(key));
}
```
and it worked out for me, but may be you can do it a better way.. 
so when can I expect this issue resolved?

**Comment by ricellis** [comment]
Hopefully soon, the fix is ready on a branch, but I'm having some issues with the Travis tests.

**Comment by ricellis** [comment]
The fix is included in the [0.4.1 release](https://github.com/cloudant/sync-cordova-plugin/releases/tag/0.4.1).

**Comment by cmijum** [comment]
Thanks a lot @ricellis Its working fine.
